pkg/registry/apis/secret/decrypt.go

```diff
@@ -9,51 +9,75 @@ import (
 	"k8s.io/apiserver/pkg/registry/rest"
 
 	secret "github.com/grafana/grafana/pkg/apis/secret/v0alpha1"
+	"github.com/grafana/grafana/pkg/setting"
 	secretstore "github.com/grafana/grafana/pkg/storage/secret"
 )
 
-type secretDecrypt struct {
-	store secretstore.SecureValueStore
-}
-
 var (
+	_ rest.Storage         = (*secretDecrypt)(nil)
+	_ rest.Scoper          = (*secretHistory)(nil)
 	_ rest.Connecter       = (*secretDecrypt)(nil)
 	_ rest.StorageMetadata = (*secretDecrypt)(nil)
 )
 
+type secretDecrypt struct {
+	config *setting.Cfg
+
+	// TODO: we can use composition and only expose the `Decrypt` method this uses.
+	store secretstore.SecureValueStore
+}
+
+// New returns an empty `*SecureValue` that is required to be implemented by any storage.
 func (r *secretDecrypt) New() runtime.Object {
 	return &secret.SecureValue{}
 }
 
-func (r *secretDecrypt) Destroy() {
+// Destroy is a no-op.
+func (r *secretDecrypt) Destroy() {}
+
+// NamespaceScoped returns `true` because the storage is namespaced (== org).
+func (r *secretDecrypt) NamespaceScoped() bool {
+	return true
 }
 
+// ConnectMethods returns the list of HTTP methods we accept for this subresource.
 func (r *secretDecrypt) ConnectMethods() []string {
-	return []string{"GET"}
+	return []string{http.MethodGet}
 }
 
+// NewConnectOptions returns some custom options that is passed in the `opts` field used by `Connect`.
 func (r *secretDecrypt) NewConnectOptions() (runtime.Object, bool, string) {
 	return nil, false, ""
 }
 
+// ProducesMIMETypes returns the `Content-Type` used by `Connect`.
 func (r *secretDecrypt) ProducesMIMETypes(verb string) []string {
 	return []string{"text/plain"}
 }
 
+// ProducesObject returns the concrete type (marshable with `json` tags) used by `Connect`.
 func (r *secretDecrypt) ProducesObject(verb string) interface{} {
+	// TODO: this could be a string, if we use `responder.Object`.
 	return &secret.SecureValue{}
 }
 
+// Connect returns an http.Handler that will handle the request/response for a given API invocation.
+// See other methods implemented for supporting/optional functionality.
 func (r *secretDecrypt) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
 	ns := request.NamespaceValue(ctx)
+
 	val, err := r.store.Decrypt(ctx, ns, name)
 	if err != nil {
 		return nil, err
 	}
+
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		if true {
-			w.WriteHeader(http.StatusOK)
-			_, _ = w.Write([]byte(val.Spec.Value)) // the raw value...
+		// !!! DANGER !!!
+		// This returns the decrypted and very raw `value` from a `securevalue`.
+		// It should not be used in production mode!
+		if r.config.Env != setting.Prod {
+			_, _ = w.Write([]byte(val.Spec.Value))
```

Comment:
By default it will have a 200 OK header when you call `w.Write`, just wanted to omit it to keep this "sensitive" part more in the front without other boilerplate.